### ISSUE TITLE
Attempting to use different AWS auth methods

### ISSUE THREAD

**Issue by Briansbum** [gruntwork-io/terratest#112]
I'm just starting to have a go at using this package and I've come across an issue with how AWS authentication is handled.

In the AWS module there are a lot of auth methods such as `NewAuthenticatedSession`, `CreateAwsSessionWithMfa` and `NewAuthenticatedSessionFromRole`. The issue is that none of these apart from `NewAuthenticatedSession` are implemented anywhere which makes them unusable.

Am I missing something or should I have a look in to how to get these properly implemented?

**Comment by brikis98** [comment]
What do you mean those methods are not implemented? They are all implemented here: https://github.com/gruntwork-io/terratest/blob/9348b5bd2fbfbe5fcf3c40b59117c16670c1f3b9/modules/aws/auth.go

**Comment by Briansbum** [comment]
Sorry I probably misspoke there.

I mean they're not available to be used in any other sections of the AWS module.

For example the S3 functions terratest/modules/aws/s3.go only make use of `NewAuthenticatedSession` with no way to use another method of authentication.

**Comment by brikis98** [comment]
Ah, I see what you mean. I can definitely see cases where you may want to pass in a custom `Session` of some sort. 

I suspect this would require a fairly large change. One way to do it that is mostly backwards compatible is to update every method that uses the AWS APIs from:

```go
func SomeMethodThatUsesAWS(t *testing.T, someParam string) {
  sess := NewAuthenticateSession()
  ec2 := ec2.New(sess)
}
```

To:

```go
func SomeMethodThatUsesAWS(t *testing.T, someParam string, sess ... *session.Session) {
  sess := NewAuthenticateSession(sess)
  ec2 := ec2.New(sess)
}
```

And to update `NewAuthenticatedSession` to use the passed in `Session`, if one is specified:

```go
func NewAuthenticatedSession(sess ... *session.Session) {
  if len(sess) > 0 {
    return sess[0]
  }

  // Otherwise, use the original logic
}
```

**Comment by Briansbum** [comment]
This looks like it would work nicely, I'll have a look at making it work.

**Comment by Briansbum** [comment]
Resolved in #119
